### PR TITLE
Fix/duplicated graphile request 2

### PR DESCRIPTION
**Pull request by vscatarino**

### CODE REVIEW

**Review by sandhilt — APPROVED** [review]
LGTM

**Review by fabiooshiro — CHANGES_REQUESTED** [review]
We still have problems:
```bash
[13:01:53.707] DBG services/service.go:107 Report exist inputIndex=2 outputIndex=2
[13:01:53.707] DBG repository/sql_executor.go:22 Using ExecContext with transaction.
[13:01:53.707] DBG repository/sql_executor.go:22 Using ExecContext with transaction.
[13:01:56.709] DBG synchronizer/graphile_fetcher.go:209 Graphile querying afterOutput="WyJwcmltYXJ5X2tleV9hc2MiLFsxLDFdXQ==" afterInput="WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=" afterReport="WyJwcmltYXJ5X2tleV9hc2MiLFsyLDJdXQ=="
[13:01:59.725] DBG synchronizer/graphile_fetcher.go:209 Graphile querying afterOutput="WyJwcmltYXJ5X2tleV9hc2MiLFsxLDFdXQ==" afterInput="WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=" afterReport="WyJwcmltYXJ5X2tleV9hc2MiLFsyLDJdXQ=="
[13:02:02.732] DBG synchronizer/graphile_fetcher.go:209 Graphile querying afterOutput="WyJwcmltYXJ5X2tleV9hc2MiLFsxLDFdXQ==" afterInput="WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=" afterReport="WyJwcmltYXJ5X2tleV9hc2MiLFsyLDJdXQ=="
^C[13:02:02.756] DBG synchronizer/graphile_synchornizer.go:61 GraphileSynchronizer canceled: Error="context canceled"
[13:02:02.757] DBG supervisor/supervisor.go:54 supervisor: worker exitted with success worker=GraphileSynchronizer
[13:02:02.757] DBG supervisor/supervisor.go:54 supervisor: worker exitted with success worker=http
[13:02:02.757] DBG supervisor/supervisor.go:54 supervisor: worker exitted with success worker=ExecListener
[13:02:02.759] DBG supervisor/supervisor.go:54 supervisor: worker exitted with success worker=anvil
[13:02:02.759] DBG supervisor/supervisor.go:81 supervisor: all workers exitted
➜  nonodo git:(fix/duplicated-graphile-request-2) ✗ export POSTGRES_HOST=127.0.0.1
export POSTGRES_PORT=5432
export POSTGRES_DB=mydatabase
export POSTGRES_USER=myuser
export POSTGRES_PASSWORD=mypassword
go run . --http-address=0.0.0.0 --high-level-graphql --enable-debug --node-version v2 --db-implementation postgres
[13:02:09.173] INF nonodo/nonodo.go:134 Using PostGres DB ...
[13:02:09.197] DBG repository/input_repository.go:57 Inputs table created
[13:02:09.197] DBG repository/report_repository.go:29 Reports table created
[13:02:09.197] DBG convenience/container.go:180 GraphileClient graphileAddress=localhost graphilePort=5001
[13:02:09.197] DBG reader/adapter_v2.go:61 NewAdapterV2
[13:02:09.197] DBG devnet/anvil.go:94 anvil: anvil is installed
[13:02:09.197] DBG nonodo/nonodo.go:182 Sync initialization
[13:02:09.198] INF nonodo/nonodo.go:225 Listening port=8080
[13:02:09.198] DBG devnet/anvil.go:132 anvil: created temp dir with state file dir=/var/folders/rv/2kdnzvbj5y77c02fpxk6hnsh0000gn/T/1094642784
[13:02:09.251] DBG supervisor/supervisor.go:59 supervisor: worker is ready worker=anvil
[13:02:09.251] DBG supervisor/supervisor.go:59 supervisor: worker is ready worker=GraphileSynchronizer
[13:02:09.251] INF convenience/voucher_exec_listener.go:84 Connecting to provider=ws://127.0.0.1:8545
[13:02:09.252] DBG supervisor/supervisor.go:59 supervisor: worker is ready worker=ExecListener
[13:02:09.252] DBG supervisor/supervisor.go:59 supervisor: worker is ready worker=http

Http Rollups for development started at http://localhost:5004
GraphQL running at http://localhost:8080/graphql
Inspect running at http://localhost:8080/inspect/
Press Ctrl+C to stop the node

[13:02:09.252] INF nonodo/main.go:485 nonodo: ready after=79.835208ms
[13:02:09.254] DBG convenience/voucher_exec_listener.go:106 ReadPastExecutions FromBlock=0
[13:02:09.256] INF convenience/voucher_exec_listener.go:166 Listening for execution events...
[13:02:09.261] DBG synchronizer/graphile_fetcher.go:209 Graphile querying afterOutput="" afterInput="WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=" afterReport="WyJwcmltYXJ5X2tleV9hc2MiLFsyLDJdXQ=="
[13:02:09.277] DBG decoder/decoder.go:72 Add Voucher/Notices inputIndex=1 outputIndex=1
[13:02:09.284] DBG repository/sql_executor.go:22 Using ExecContext with transaction.
[13:02:09.284] DBG repository/sql_executor.go:22 Using ExecContext with transaction.
[13:02:12.287] DBG synchronizer/graphile_fetcher.go:209 Graphile querying afterOutput="WyJwcmltYXJ5X2tleV9hc2MiLFsxLDFdXQ==" afterInput="WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=" afterReport="WyJwcmltYXJ5X2tleV9hc2MiLFsyLDJdXQ=="
[13:02:15.295] DBG synchronizer/graphile_fetcher.go:209 Graphile querying afterOutput="WyJwcmltYXJ5X2tleV9hc2MiLFsxLDFdXQ==" afterInput="WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=" afterReport="WyJwcmltYXJ5X2tleV9hc2MiLFsyLDJdXQ=="
[13:02:18.310] DBG synchronizer/graphile_fetcher.go:209 Graphile querying afterOutput="WyJwcmltYXJ5X2tleV9hc2MiLFsxLDFdXQ==" afterInput="WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=" afterReport="WyJwcmltYXJ5X2tleV9hc2MiLFsyLDJdXQ=="
[13:02:21.321] DBG synchronizer/graphile_fetcher.go:209 Graphile querying afterOutput="WyJwcmltYXJ5X2tleV9hc2MiLFsxLDFdXQ==" afterInput="WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=" afterReport="WyJwcmltYXJ5X2tleV9hc2MiLFsyLDJdXQ=="
^C[13:02:22.589] DBG synchronizer/graphile_synchornizer.go:61 GraphileSynchronizer canceled: Error="context canceled"
[13:02:22.589] DBG supervisor/supervisor.go:54 supervisor: worker exitted with success worker=GraphileSynchronizer
[13:02:22.589] DBG supervisor/supervisor.go:54 supervisor: worker exitted with success worker=http
[13:02:22.590] DBG supervisor/supervisor.go:54 supervisor: worker exitted with success worker=ExecListener
[13:02:22.592] DBG supervisor/supervisor.go:54 supervisor: worker exitted with success worker=anvil
[13:02:22.592] DBG supervisor/supervisor.go:81 supervisor: all workers exitted

```

File: internal/convenience/synchronizer/graphile_synchornizer.go [hunk 1]
```diff
@@ -196,12 +197,12 @@ func (x GraphileSynchronizer) handleGraphileResponse(ctx context.Context, output
 	synchronizeFetch := &model.SynchronizerFetch{
 		TimestampAfter:       uint64(time.Now().UnixMilli()),
 		IniCursorAfter:       initCursorAfter,
-		EndCursorAfter:       x.GraphileFetcher.CursorAfter,
+		EndCursorAfter:       outputResp.Data.Outputs.PageInfo.EndCursor,
```
Comment (fabiooshiro) [inline internal/convenience/synchronizer/graphile_synchornizer.go:200]: @vscatarino here

**Review by fabiooshiro — APPROVED** [review]

**Comment by fabiooshiro** [comment]
it's working!!!
I also tested the database transaction rollback, and it worked perfectly.

good job! @vscatarino & @DiegoSena

**Comment by fabiooshiro** [comment]
@sandhilt could you help us?
image

**Pull request merged**